### PR TITLE
Fix "index out of range" error while getting IP

### PR DESCRIPTION
**Pull request by carlosedp**
## Description

When getting IP for the created container, if the host does not has eth0
as the default network interface, the "index out of range" error is
generated.
This gets the correct interface for the namespace of the container PID.

## Motivation and Context

- [ ] I have raised an issue to propose this change **this is required**

This fixes the error on issue #24 .

## How Has This Been Tested?

Tested on a host where the default network interface is `eth1` that generated the reported error. With this fix, the IP is returned correctly.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Commits:

- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] My commit message has a body and describe how this was tested and why it is required.
- [x] I have signed-off my commits with `git commit -s` for the Developer Certificate of Origin (DCO)

Code:

- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.

Docs:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: handlers/deploy.go [hunk 1]
```diff
@@ -150,23 +150,16 @@ func deploy(ctx context.Context, req types.FunctionDeployment, client *container
...
-	}
-	if result.Interfaces[defaultInterface].IPConfigs != nil &&
-		len(result.Interfaces[defaultInterface].IPConfigs) == 0 {
-		return fmt.Errorf("failed to find IP for interface %q, no configs found", defaultInterface)
+	// Get the IP of the created interface
+	for _, config := range result.Interfaces {
+		if config.Sandbox == netns {
+			for _, ipConfig := range config.IPConfigs {
+				serviceMap.Add(name, &ipConfig.IP)
+				log.Printf("%s has IP: %s.\n", name, &ipConfig.IP)
```
Comment (alexellis) [inline handlers/deploy.go:158]: Wait, why aren't we doing a break out of the loops here?

File: handlers/deploy.go [hunk 2]
```diff
@@ -150,23 +150,16 @@ func deploy(ctx context.Context, req types.FunctionDeployment, client *container
...
+			}
+		}
```
Comment (alexellis) [inline handlers/deploy.go:160]: Let's discover the IP in the loop, then act on it after the loop, either from breaking out or from finishing without a result.

**Review by alexellis — CHANGES_REQUESTED** [review]
The network discovery could be improved. Please can you extract it to its own function before merge.

**Comment by carlosedp** [comment]
This has been replaced by the refactor on PR #26

**Pull request closed without merging**